extensions/vertx/deployment/src/main/java/io/quarkus/vertx/deployment/VertxProcessor.java

```diff
@@ -329,4 +342,69 @@ private void invoke(BeanInfo bean, MethodInfo method, ResultHandle messageHandle
+        Collection<AnnotationInstance> consumeEventAnnotationInstances = index.getAnnotations(CONSUME_EVENT);
+        Set<Type> returnTypes = new HashSet<>();
+        for (AnnotationInstance consumeEventAnnotationInstance : consumeEventAnnotationInstances) {
+            AnnotationTarget typeTarget = consumeEventAnnotationInstance.target();
+            if (typeTarget.kind() != AnnotationTarget.Kind.METHOD) {
+                continue;
+            }
+
+            Type returnType = typeTarget.asMethod().returnType();
+            Type typeToAdd = null;
+            if (returnType.kind() == Kind.CLASS) {
+                typeToAdd = returnType;
+            } else if (returnType.kind() == Kind.PARAMETERIZED_TYPE) {
+                ParameterizedType returnedParamType = returnType.asParameterizedType();
+
+                if (returnedParamType.name().equals(MESSAGE)) {
+                    typeToAdd = returnedParamType.arguments().get(0);
+                } else {
+                    typeToAdd = returnedParamType;
+                }
+            }
+            if (typeToAdd == null) {
+                LOGGER.warnf(
+                        "Cannot load return type of the consumer event annoted method %s",
+                        typeTarget);
+                continue;
+            }
+            returnTypes.add(typeToAdd);
+            AnnotationValue codec = consumeEventAnnotationInstance.value("codec");
+            if (codec != null && codec.asClass().kind() == Kind.CLASS) {
+                registerCodec(typeToAdd, codec.asClass().asClassType().name());
+                continue;
+            }
+
+            registerCodec(typeToAdd, GENERIC_MESSAGE_CODEC);
+        }
+    }
+
+    private void registerCodec(Type typeToAdd, DotName codec) {
+        try {
+            Class<?> messageCodecClass = Class.forName(codec.toString());
+            if (messageCodecClass.isAssignableFrom(MessageCodec.class)) {
+                MessageCodec messageCodec = (MessageCodec) messageCodecClass.newInstance();
+                registerCodec(typeToAdd, messageCodec);
+            } else {
+                LOGGER.warn(String.format("The codec %s does not inherite from MessageCodec ", codec.toString()));
+            }
+        } catch (ClassNotFoundException e) {
+            LOGGER.warn("Cannot instantiate the MessageCodec " + codec.toString() + " because " + e.getMessage());
+        } catch (InstantiationException e) {
+            LOGGER.warn("Cannot instantiate the MessageCodec " + codec.toString() + " because " + e.getMessage());
+        } catch (IllegalAccessException e) {
+            LOGGER.warn("Cannot instantiate the MessageCodec " + codec.toString() + " because " + e.getMessage());
+        }
+
+    }
+
+    private void registerCodec(Type typeToAdd, MessageCodec codec) {
+        final Vertx vertx = Vertx.vertx();
+        EventBus eventBus = vertx.eventBus();
+        eventBus.registerDefaultCodec(typeToAdd.name().getClass(), codec);
```

Comment:
you are not registering the codec on the right event bus. They need to be registered on the Vert.x instance that is injected.